crates/api/src/publish.rs

```diff
@@ -0,0 +1,131 @@
+//! Kitsune2 publish types.
+
+use crate::{builder, config};
+use crate::{
+    transport::DynTransport, BoxFut, DynFetch, K2Result, OpId, SpaceId, Url,
+};
+use bytes::{Bytes, BytesMut};
+use prost::Message;
+use std::sync::Arc;
+
+pub(crate) mod proto {
+    include!("../proto/gen/kitsune2.publish.rs");
+}
+
+pub use proto::{k2_publish_message::*, K2PublishMessage, PublishOps};
+
+impl From<Vec<OpId>> for PublishOps {
+    fn from(value: Vec<OpId>) -> Self {
+        Self {
+            op_ids: value.into_iter().map(Into::into).collect(),
+        }
+    }
+}
+
+impl From<PublishOps> for Vec<OpId> {
+    fn from(value: PublishOps) -> Self {
+        value.op_ids.into_iter().map(Into::into).collect()
+    }
+}
+
+/// Serialize list of op ids to request.
+pub fn serialize_publish_ops(value: Vec<OpId>) -> Bytes {
+    let mut out = BytesMut::new();
+    PublishOps::from(value)
+        .encode(&mut out)
+        .expect("failed to encode publish ops request");
+    out.freeze()
+}
+
+/// Serialize list of op ids to fetch request message.
+pub fn serialize_publish_ops_message(value: Vec<OpId>) -> Bytes {
+    let mut out = BytesMut::new();
+    let data = serialize_publish_ops(value);
+    let publish_message = K2PublishMessage {
+        publish_message_type: PublishMessageType::Ops.into(),
+        data,
+    };
+    publish_message
+        .encode(&mut out)
+        .expect("failed to encode publish ops message");
```

Comment:
Just noticing that the fetch module does this too. I'd prefer bubbling this error up if something goes wrong. This isn't very likely to fail but I'd prefer to minimize panics

Can this be changed in the fetch module too @jost-s or did you have a reason for `expect`ing these?

Reply:
This can only fail in case of a memory corruption. There was a conversation with David over how to handle this on some PR. He left it up to me. If there's something wrong with the memory, I think it's justified to panic.

Reply:
Quoting from the prost docs:
> EncodeError always indicates that a message failed to encode because the provided buffer had insufficient capacity. Message encoding is otherwise infallible.